### PR TITLE
fix: app crash fix

### LINKED ISSUE #101: bug: app crashes if the node is not up or if the provided ln socket path is not correct.
Screenshot for the error: 

image

### PR DESCRIPTION
**Pull request by Harshit933**
fixes #101 

Some screenshots: 

When we provide wrong parameters:
image

When the node is not up: 
image

Error handling could be improved. This can be the starting point for it.

### CODE REVIEW

File: lib/views/home/home_view.dart [hunk 1]
```diff
@@ -201,8 +201,13 @@ class _HomeViewState extends State<HomeView> {
     return FutureBuilder<List<dynamic>?>(
         future: listPayments(),
         builder: (context, AsyncSnapshot<List<dynamic>?> snapshot) {
-          _checkIfThereAreError(context: context, snapshot: snapshot);
-          if (snapshot.hasData) {
+          String error = '';
```
Comment (vincenzopalazzo) [inline lib/views/home/home_view.dart:204]:
```suggestion
          String? error = null;
```

and why declare this outside of the if? is it not used to store the error message?

Comment (vincenzopalazzo) [reply lib/views/home/home_view.dart:204]: btw, no empty string, but use the nullable operator, let the language fearture help you.

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: lib/views/home/home_view.dart (hunk 1 above)
Comment (Harshit933) [reply lib/views/home/home_view.dart:204]: > and why declare this outside of the if? is it not used to store the error message?

Yes it should be inside.Fixing it now.

File: lib/views/home/home_view.dart [hunk 2]
```diff
@@ -330,7 +329,7 @@ class _HomeViewState extends State<HomeView> {
                   child: Padding(
                     padding: const EdgeInsets.symmetric(horizontal: 10),
                     child: Text(
-                      error,
+                      error!,
```
Comment (vincenzopalazzo) [inline lib/views/home/home_view.dart:332]: This will crash if the code goes in the `else if (snapshot.hasData) {` because  `error` is null, right?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: lib/views/home/home_view.dart (hunk 2 above)
Comment (Harshit933) [reply lib/views/home/home_view.dart:332]: Oh I got this error. This error occurs at an instance when we are fetching the data from the server. This condition hits then only. I have a fix for this.

**Review by vincenzopalazzo — APPROVED** [review]

**Comment by vincenzopalazzo** [comment]
Please take in mind that we are not allow merge commit inside the history, next time try to rebase on top of the master

**Pull request merged**